rust/core-lib/src/editor.rs

```diff
@@ -554,13 +554,26 @@ impl Editor {
         tab_text
     }
 
-    // TODO: insert from keyboard or input method shouldn't break undo group,
-    // but paste should.
     fn do_insert(&mut self, view: &View, chars: &str) {
         self.this_edit_type = EditType::InsertChars;
         self.insert(view, chars);
     }
 
+    fn do_paste(&mut self, view: &View, chars: &str) {
+        if view.sel_regions().len() == 1
```

Comment:
This means that it will get an EditType of Other? That seems right to me, but want to think it through.

Reply:
Yes, it gets `Other`; previously it was `Insert`.